cluster-autoscaler/cloudprovider/aws/aws_cloud_provider_test.go

```diff
@@ -276,3 +300,8 @@ func TestBuildAsg(t *testing.T) {
 	assert.Equal(t, 222, asg.MaxSize())
 	assert.Equal(t, "test-name", asg.Name)
 }
+
+func TestAutoDiscoverASG(t *testing.T) {
+	provider := testProvider(t, testAwsManager)
+	provider.autoDiscoverASG()
```

Comment:
I don't see how this is much of a test, we should be asserting that `autoDiscoverASG` returns nil and that the appropriate ASGs were registered